doc/userguide/devguide/contributing/code-submission-process.rst

```diff
@@ -9,16 +9,18 @@ Commits
 #. Commits need to be logically separated. Don't fix unrelated things in one commit.
 #. Don't add unnecessary commits, if commit 2 fixes commit 1 merge them together (squash)
 #. Commits need to have proper messages, explaining anything that is non-trivial
-#. Commits should not at the same time change, rename and/or move code. Use separate commits
-   for each of this, e.g, a commit to rename files, then a commit to change the code.
-#. Documentation updates should be in their own commit (not mixed with code commits)
-#. Commit messages need to be properly formatted:
-    * Meaningful and short (50 chars max) subject line followed by an empty line
-    * Naming convention: prefix message with sub-system ("rule parsing: fixing foobar"). If
-      you're not sure what to use, look at past commits to the file(s) in your PR.
-    * Description, wrapped at ~72 characters
+#. Commits should not, at the same time, change, rename and/or move code. Use separate commits
+     for each of this, e.g, a commit to rename files, then a commit to change the code.
+#. If your code changes or adds new behavior, update the documentation in the same commit.
+     Otherwise, documentation updates should be in their own commit (not mixed with code commits).
+#. Commit messages need to be properly formatted (check the example further
```

Comment:
we have a separate doc for this?

Reply:
From memory, and my checks, so far, this has been the place where we mentioned this stuff. Want me to break it into another document?

Reply:
no I think this is the doc I remembered, just not realized we were talking about it now ;)